core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
If there is a way to set the pool property in hive thriftserver then we should honour the same instead of adding a separate one. Looks like it was added here: https://github.com/apache/spark/pull/1937/files

Note that the hive-thriftserver can run both in embedded mode (was required for 1.0 but deferred for 1.1) or in split mode so two different SQL properties will confuse users. Since our JDBC/ODBC server is also a thrift server, so that name should be fine.